lib/Value/Matrix.pm

```diff
@@ -291,6 +291,119 @@ sub isZero {
 	return 1;
 }
 
+#
+#  See if the matrix is triangular, diagonal, symmetric, orthogonal
+#
+
+sub isUpperTriangular {
+	my $self = shift;
+	my @d    = $self->dimensions;
+	return 1 if scalar(@d) == 1;
+	return 0 if scalar(@d) > 2;
+	for my $i (2 .. $d[0]) {
+		for my $j (1 .. min($i - 1, $d[1])) {
```

Comment:
There is not `min` method in the `Value::Matrix` package.  So this causes an error.  I am not sure that there is a `min` method to pull in, so you might need to implement it directly.  You could change this to
```suggestion
		for my $j (1 .. ($i - 1 < $d[1] ? $i - 1 : $d[1])) {
```